### ISSUE TITLE
`CONTRIBUTING.md` doesn't really

### ISSUE THREAD

**Issue by filmil** [rust-lang/arewewebyet#248]
Hi folks.  Noting this user experience down, so I don't forget. 

tl;dr: The doc piece `CONTRIBUTING.md` does not really explain how to contribute, but it should. Examples below, in no specific order.

It says:

> If a package isn't listed on the website yet, feel free to send us a pull-request for it.

Which is cool, and I'd be glad to do that, but as it is, I need to puzzle out what you want me to do to achieve that. It would be more useful to have a checklist-like sequence of steps with instructions on what to do when:

- I want to add a new package
- I want to modify an existing package
- I want to submit project news.

Would also be nice to have pointers to commits that achieve these goals.  For example, it is unclear if you want me to regenerate html files when adding a package or not. Is adding the JSON metadata enough?

Early on, the file says:

> Please refer to the CONTRIBUTING document for details.

But this *is* that document.  No need for another level of indirection.

> Please make sure the pull request contains the following:
> - the data for your package (see the [DOCS] on how to do that)
> - adds the package to the appropriate topic(s)

But, the DOCS file doesn't really explain how to do that, it is a laundry list of things you may want to do.  There should be instructions here on how to prepare a pull request in the form that is acceptable to the project.

In general, try to replace prose with checklists and bullet points that are easy to follow.

HTH. I'll fix what and where I can.

**Comment by ghost** [comment]
Great points! 
> Please refer to the CONTRIBUTING document for details.

I removed the self-referential line in #278.

I feel like adding how-to guides would address some of the confusion @filmil notes whilst increasing the likelihood new contributors will have a positive experience. 

I have been parroting @evildmp's fourfold documentation structure: https://documentation.divio.com/introduction/#the-problem-and-the-solution. It works well. How-to guides provide clear steps on a specific task. 


I propose we create a single how-to for each task:

> 
>    I want to add a new package
>    I want to modify an existing package
>   I want to submit project news.

**Comment by ibraheemdev** [comment]
@efx @filmil Now that #293 was merged, remote data is fetched during the build, meaning that we don't have to deal with `.json` data files and scripts, so contributing has become much easier. I removed the `DOCS` page, and updated the readme and `CONTRIBUTING`. Can this be closed, or are there any more concerns?
